examples/transactions.c

```diff
@@ -321,112 +293,125 @@ static void commit_transaction_and_start_new (struct state *state) {
-        error = rd_kafka_begin_transaction(state->producer);
+        error = rd_kafka_begin_transaction(producer);
         if (error)
                 fatal_error("Failed to begin new transaction", error);
 }
 
 /**
  * @brief The rebalance will be triggered (from rd_kafka_consumer_poll())
  *        when the consumer's partition assignment is assigned or revoked.
- *
- * Prior to KIP-447 being supported there must be one transactional output
- * producer for each consumed input partition, so we create and destroy
- * these producer's from this callback.
  */
 static void
-consumer_group_rebalance_cb (rd_kafka_t *rk,
+consumer_group_rebalance_cb (rd_kafka_t *consumer,
                              rd_kafka_resp_err_t err,
                              rd_kafka_topic_partition_list_t *partitions,
                              void *opaque) {
-        int i;
+        rd_kafka_t *producer = (rd_kafka_t *)opaque;
+        rd_kafka_error_t *error;
 
-        if (!strcmp(rd_kafka_rebalance_protocol(rk), "COOPERATIVE"))
-                fatal("This example has not yet been modified to work with "
-                      "cooperative incremental rebalancing "
+        if (strcmp(rd_kafka_rebalance_protocol(consumer), "COOPERATIVE"))
+                fatal("This example requires cooperative incremental "
+                      "rebalancing "
                       "(partition.assignment.strategy=cooperative-sticky)");
 
         switch (err)
         {
         case RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS:
-                assigned_partitions =
-                        rd_kafka_topic_partition_list_copy(partitions);
-
-                fprintf(stdout, "Consumer group rebalanced, new assignment:\n");
-
-                /* Create a transactional producer for each input partition */
-                for (i = 0 ; i < assigned_partitions->cnt ; i++) {
-                        /* Store the partition-to-producer mapping
-                         * in the partition's opaque field. */
-                        rd_kafka_topic_partition_t *rktpar =
-                                &assigned_partitions->elems[i];
-                        struct state *state = calloc(1, sizeof(*state));
-
-                        state->producer = create_transactional_producer(rktpar);
-                        state->rktpar = rktpar;
-                        rktpar->opaque = state;
-                        state->last_commit = time(NULL);
-
-                        fprintf(stdout,
-                                " %s [%d] with transactional producer %s\n",
-                                rktpar->topic, rktpar->partition,
-                                rd_kafka_name(state->producer));
-                }
+                fprintf(stdout,
+                        "Consumer group rebalanced: "
+                        "%d new partition(s) assigned\n",
+                        partitions->cnt);
 
                 /* Let the consumer know the rebalance has been handled
```

Comment:
i'm not into the words "let the consumer know". i'd say what it does (assign the consumer to read from the additional partitions), and that it's mandatory.